pages/ekmann.js

```diff
@@ -0,0 +1,352 @@
+        if (!value) {
+          // If any field is empty, log error and exit early
+          setEmptyFieldsError(
+            `Field '${key}' is empty. Please enter some text in "EVERY" field.`
+          );
+
+          return;
+        }
+      }
+    }
+    event.preventDefault();
+    //here you can log the data which is saved before it i processed very well
+    const inputText = JSON.stringify(formData);
+
+    const url =
+      "https://ekman-emotion-analysis.p.rapidapi.com/ekman-emotion?all=true";
+    //?all=true for all emotions without filter
+    const options = {
+      method: "POST",
+      headers: {
+        "content-type": "application/json",
+        Accept: "application/json",
+        "X-RapidAPI-Key": "d494478f31mshab8f5690c7d60c8p1f2035jsna0a331ad02cd",
+        "X-RapidAPI-Host": "ekman-emotion-analysis.p.rapidapi.com",
+      },
+      body: JSON.stringify([
+        {
+          id: "1",
+          language: language,
+          text: inputText,
+        },
+      ]),
+    };
+    // Reset form after submission (optional)
+    setFormData({
+      0: "",
+      1: "",
+      2: "",
+      3: "",
+      4: "",
+      5: "",
+      6: "",
+      7: "",
+      8: "",
+      9: "",
+    });
+    try {
+      const response = await fetch(url, options);
+      const result = await response.json();
+      //.text(); makes the response a string - to read the response body as plain text
+      setEmotionResult(result);
+      setPredictionsState(true);
+    } catch (error) {
+      console.error(error);
+    }
+  };
+
+  const question = [
+    "1. What word describes your primary emotion right now?",
+    "2. How would you describe that emotion?",
+    "3. How does your body feel?",
+    "4. Is there an event, person, place, or thing that might have caused this response?",
+    "5. How did you react to that trigger?",
+    "6. How will you respond now?",
+    "7. Are you taking care of your physical well-being?",
+    "8. What is on your mind today?",
+    "9. What can you do to feel calmer?",
+    "10. What are you grateful for today?",
+  ];
+
+  const predictions =
+    emotionResult && emotionResult[0] && emotionResult[0].predictions;
+
+  const transformedPredictions =
+    predictions &&
+    predictions[0] &&
+    predictions.map((prediction) => {
+      if (prediction.prediction === "noemo") {
+        return { ...prediction, prediction: "no-emotion" };
+      } else {
+        return prediction;
+      }
+    });
+
+  function getColorForEmotion(emotion) {
+    switch (emotion) {
+      case "joy":
+        return "yellow";
+      case "surprise":
+        return "orange";
+      case "sadness":
+        return "blue";
+      case "noemo":
+        return "gray";
+      case "fear":
+        return "purple";
+      case "anger":
+        return "red";
+      case "disgust":
+        return "green";
+      default:
+        return "lightgrey"; // Default color for unknown emotions
+    }
+  }
```

Comment:
this is definitely something that you should outsource into another "utils" file or folder, as I have seen this in other places in your app. Make it as reusable as you can!

Reply:
I need more colors, cause I have different values as the other emotion-color-analyzer. I would let it in the ekmann.js for now.